### PR TITLE
bt-433:  Connect get user data API to mobile and remove related TODOs

### LINKED ISSUE #433: Connect get user data API to mobile and remove related TODOs
[http://bsa-talents.us-east-1.elasticbeanstalk.com/api/v1/documentation/static/index.html#/User%20Details/get_user_details__userId_](http://bsa-talents.us-east-1.elasticbeanstalk.com/api/v1/documentation/static/index.html#/User%20Details/get_user_details__userId_)

### PR DESCRIPTION
**Pull request by likeri29**
.

### CODE REVIEW

File: mobile/src/bundles/talent/talent-api.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { ApiPath, ContentType } from '~/bundles/common/enums/enums';
 import { UserDetailsApiPath } from '~/bundles/talent/enums/enums';
 import {
     type UserDetailsCreateRequestDto,
-    type UserDetailsFindRequestDto,
+    // type UserDetailsFindRequestDto,
```
Comment (nikita-remeslov) [inline mobile/src/bundles/talent/talent-api.ts]: ?

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: mobile/src/bundles/talent/talent-api.ts (hunk 1 above)
Comment (likeri29) [reply mobile/src/bundles/talent/talent-api.ts]: Will delete it

File: mobile/src/bundles/talent/screens/profile/profile.tsx [hunk 2]
```diff
@@ -57,6 +57,7 @@ const Profile: React.FC = () => {
         <View style={globalStyles.flex1}>
             <NewAccountHeader title={stepTitle} currentStep={stepNumber} />
             <ProfileForm
+                key={profileStepData?.profileName}
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/screens/profile/profile.tsx]: should be replaced with some form reinitialize

Comment (katerynakostikova) [reply mobile/src/bundles/talent/screens/profile/profile.tsx]: or set initial form values only when data fully loaded

File: mobile/src/bundles/talent/store/actions.ts [hunk 3]
```diff
@@ -58,13 +57,16 @@ const updateOnboardingData = createAsyncThunk<
...
 
 const getTalentDetails = createAsyncThunk<
-    UserDetailsResponseDto | null,
-    UserDetailsFindRequestDto,
+    UserDetailsGeneralResponseDto | null,
+    UserDetailsGeneralResponseDto,
     AsyncThunkConfig
->(`${sliceName}/getTalentDetails`, (detailsPayload, { extra }) => {
-    const { talentApi, notifications } = extra;
+>(`${sliceName}/getTalentDetails`, async (payload, { extra }) => {
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/store/actions.ts]: this endpoint will be used both for talent and employer

**Review by katerynakostikova — CHANGES_REQUESTED** [review]

File: mobile/src/bundles/talent/store/actions.ts (hunk 3 above)
Comment (likeri29) [reply mobile/src/bundles/talent/store/actions.ts]: If you mean getUserDetailsById yes, we will use it for employer too, but do we need create new class only for this request?

File: mobile/src/bundles/talent/store/actions.ts [hunk 4]
```diff
@@ -58,13 +57,16 @@ const updateOnboardingData = createAsyncThunk<
 });
```
Comment (anogire) [inline mobile/src/bundles/talent/store/actions.ts:55]: `return stepPayload as unknown as UserDetailsGeneralResponseDto;`
please, delete this 'unknown' from previous PR.

Comment (likeri29) [reply mobile/src/bundles/talent/store/actions.ts:55]: I tried, but if I delete it get an error

File: mobile/src/bundles/talent/store/actions.ts (hunk 3 above)
Comment (katerynakostikova) [reply mobile/src/bundles/talent/store/actions.ts]: It shouldn't be named `getTalentDetails`

Comment (katerynakostikova) [reply mobile/src/bundles/talent/store/actions.ts]: we'll use this action for both roles

File: mobile/src/bundles/talent/store/actions.ts (hunk 4 above)
Comment (katerynakostikova) [reply mobile/src/bundles/talent/store/actions.ts:55]: which error

Comment (likeri29) [reply mobile/src/bundles/talent/store/actions.ts:55]: Everything is good now

**Review by anogire — APPROVED** [review]

**Review by katerynakostikova — APPROVED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**